### ISSUE TITLE
Add sjcl client-side crypto

### ISSUE THREAD

**Issue by fincham** [appsattic/paste.gd#1]
Steal the implementation from https://paste.nothing.net.nz/

I'm of the general mind that to keep things simple you should just encrypt every paste. But if you want to make it optional the frontend or backend can easily detect whether a paste is encrypted or not and act appropriately, no real need for a separate URL namespace (I don't think anyway).

**Comment by fincham** [comment]
Only improvement over paste.nothing.net.nz that I would float is to bring in (or just implement) an external dependency on a better encoder than base64 for the key. One of Phil Zimmerman's encoding formats like z-base-32 would be a personal favourite (https://philzimmermann.com/docs/human-oriented-base-32-encoding.txt)

**Comment by chilts** [comment]
Yes, fair enough, the server knows if it's encrypted or not. I'm happy to leave the encryption decision to the user. There are some pastes which don't matter, and perhaps you want them indexed in Google or otherwise.

Orthogonal options would be:

* public / unlisted (ie. I may do a 'latest pastes' list, or not)
* encrypted (implies unlisted)
* delete after "one read" / "specific timespan" / "never"

My plan (if I needed it) was to use Vue.js for this project, but I'm happy with whatever you think might work. Also, I usually try and load JS libs from CDNJS e.g. https://cdnjs.com/libraries/sjcl because I can't be bothered vendoring them and changing them to keep them up to date.

(Feel free to ignore any and all of this.)

**Comment by fincham** [comment]
Jolly good. I'll have a whack at it over the weekend.
